### PR TITLE
Consolidate gendocs and genkubedocs commands

### PR DESCRIPTION
**Pull request by mkumatag**
**What this PR does / why we need it**:
I see repeated code under both `cmd/gendocs` and `cmd/genkubedocs` for doing same job, so this PR will consolidate both of them into one common command  `cmd/gendocs`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```NONE
```

### CODE REVIEW

**Comment by mkumatag** [comment]
/sig/docs

File: cmd/gendocs/gen_kube_docs.go [hunk 1]
```diff
@@ -73,6 +76,10 @@ func main() {
 		// generate docs for kubelet
 		kubelet := kubeletapp.NewKubeletCommand()
 		doc.GenMarkdownTree(kubelet, outDir)
+	case "kubectl":
```
Comment (xiangpengzhao) [inline cmd/gendocs/gen_kube_docs.go]: I don't know if this is what we expected when we put kubectl into separate repo.

@shiywang

File: cmd/gendocs/gen_kubectl_docs.go [hunk 2]
```diff
@@ -1,52 +0,0 @@
...
-
-	outDir, err := genutils.OutDir(path)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed to get output directory: %v\n", err)
-		os.Exit(1)
-	}
-
-	// Set environment variables used by kubectl so the output is consistent,
-	// regardless of where we run.
-	os.Setenv("HOME", "/home/username")
```
Comment (shiywang) [inline cmd/gendocs/gen_kubectl_docs.go]: this env still in use in swagger ? why set to "/home/username" ? @pwittrock

File: cmd/gendocs/gen_kube_docs.go (hunk 1 above)
Comment (shiywang) [inline cmd/gendocs/gen_kube_docs.go]: I'm ok with this change, but I'll let @pwittrock to chime in

File: cmd/gendocs/gen_kubectl_docs.go (hunk 2 above)
Comment (mkumatag) [inline cmd/gendocs/gen_kubectl_docs.go]: ping @pwittrock

**Comment by mkumatag** [comment]
ping @jbeda @pwittrock

**Comment by steveperry-53** [comment]
@mkumatag I don't understand what's going on in this PR. On the one hand, it looks like you are removing the code under cmd/gendocs and keeping the code under cmd/genkubedocs. On the other hand, it looks like you are doing the reverse. Example: in hack/generate-docs.sh, you remove the line "cmd/genkubedocs".

Also, it's my understanding that the kubectl reference documentation is now generated by the code at [kubernetes-incubator/reference-docs/gen-kubectldocs](https://github.com/kubernetes-incubator/reference-docs/tree/master/gen-kubectldocs).

**Pull request closed without merging**